suite-native/device/src/middlewares/deviceConnectionMiddleware.ts

```diff
@@ -0,0 +1,96 @@
+import {
+    Dispatch,
+    ListenerEffectAPI,
+    UnknownAction,
+    createListenerMiddleware,
+} from '@reduxjs/toolkit';
+
+import {
+    deviceConnectThunks,
+    selectIsDeviceConnectedAndAuthorized,
+    selectIsDeviceInitialized,
+    selectIsDeviceRemembered,
+    selectIsDeviceUsingPassphrase,
+} from '@suite-common/wallet-core';
+import { selectIsFirmwareInstallationRunning } from '@suite-native/firmware';
+import {
+    AuthorizeDeviceStackRoutes,
+    RootStackRoutes,
+    checkIsRouteAccessAllowed,
+    navigationContainerRef,
+} from '@suite-native/navigation';
+import { selectIsCoinEnablingInitFinished } from '@suite-native/settings';
+
+import { NativeDeviceRootState, selectIsDeviceCompromised } from '../selectors';
+
+const DEVICE_CONNECTION_BLACKLISTED_ROUTES = [
+    RootStackRoutes.DeviceCompromisedModal,
+    RootStackRoutes.DeviceOnboardingStack,
+];
+
+export const deviceConnectionMiddleware = createListenerMiddleware<NativeDeviceRootState>();
+
+const connectDevice = ({ isCoinEnablingInitFinished }: { isCoinEnablingInitFinished: boolean }) => {
+    // If coin enabling is not finished, it takes priority over connecting screen
+    if (isCoinEnablingInitFinished) {
+        navigationContainerRef.navigate(RootStackRoutes.AuthorizeDeviceStack, {
+            screen: AuthorizeDeviceStackRoutes.ConnectingDevice,
+        });
+    } else {
+        navigationContainerRef.navigate(RootStackRoutes.CoinEnablingInit);
+    }
+};
+
+deviceConnectionMiddleware.startListening({
+    predicate: (action, currentState) =>
+        deviceConnectThunks.fulfilled.match(action) &&
+        // TODO this should dissappear after we merge device onboarding redirect here as well
+        // https://github.com/trezor/trezor-suite/issues/20157
+        // If device is not initialized and is compromised, we display the modal (reason why this condition is here) and then want to redirect to uninitialized device landing.
+        (selectIsDeviceInitialized(currentState) || selectIsDeviceCompromised(currentState)),
+    effect: (
+        _action: UnknownAction,
+        { getState }: ListenerEffectAPI<NativeDeviceRootState, Dispatch<UnknownAction>>,
+    ) => {
+        const shouldNavigateToDeviceCompromisedModal = selectIsDeviceCompromised(getState());
+        const isCoinEnablingInitFinished = selectIsCoinEnablingInitFinished(getState());
+
+        const isDeviceConnectionNavigationAllowed = checkIsRouteAccessAllowed({
+            blacklist: selectIsDeviceRemembered(getState())
```

Comment:
Why is there a condition for device.remember for SendStack?

Reply:
You can access send flow with view only device and start creating transaction without connected device. If you connect device, we block the navigation to connecting screen and stay on send stack. 

This is how it was in `useHandleDeviceConnection` so I kept the functionality. Do you think it's worth a comment here explaining why?

Reply:
After discussing, we don't have to use the device.remember and just always add the route to the blacklist.
https://github.com/trezor/trezor-suite/pull/20081/commits/65276fbeb3905ea71ae2760b1fb436dea9c7ca47